script/tool/lib/src/common/package_looping_command.dart

```diff
@@ -45,7 +84,7 @@ abstract class PackageLoopingCommand extends PluginCommand {
   /// be included in the final error summary (e.g., a command that only has a
   /// single failure mode), or strings that should be listed for that package
   /// in the final summary. An empty list indicates success.
-  Future<List<String>> runForPackage(Directory package);
+  Future<PackageResult> runForPackage(Directory package);
```

Comment:
Nice improvement.

Reply:
Yes, the fact that I had to give the magic values `[]` and `['']` special names in the original API should have been more of a red flag to me than it was.